lib/kinesis_client.rb

```diff
@@ -90,21 +91,22 @@ def push_batch(batch)
   end
 
   def push_records
-    if @records.length > 0 && @records.length < @batch_size
-      push_batch(@records)
-    else
-      @records.each_slice(@batch_size) { |slice| push_batch(slice) }
+    if @records.length > 0
+      if @records.length < @batch_size
+        push_batch(@records)
+      else
```

Comment:
I don't think this nested if is necessary. If `@records.length < @batch_size`, `each_slice` will only have one slice anyway.